File: 144. Binary Tree Preorder Traversal.py
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        
        stack = [root]
        res = []
        while stack:
            node = stack.pop()
            res.append(node.val)
            if node.right:
                stack.append(node.right)
            if node.left:
                stack.append(node.left)

        return res

File: 144. Binary Tree Preorder Traversal.go
func preorderTraversal(root *TreeNode) []int {
	if root == nil{return []int{}}
    var res []int;
	stack := []*TreeNode{root}

	for len(stack) > 0{
		node := stack[len(stack) - 1]
		stack := stack[1:]
		res = append(res, node.Val)
		if node.Right != nil{stack = append(stack, node.Right)}
		if node.Left != nil{stack = append(stack, node.Left)}
	
	}

	return res;
}
